app/models/container_deployment_authentication.rb

```diff
@@ -0,0 +1,5 @@
+class ContainerDeploymentAuthentication < ApplicationRecord
```

Comment:
Honestly, I feel like these would be better in the `authentications` table.

It has the `AuthenticationMixin` to deal with logic specific to the different types of fields that are embodied there already.  It's already used for storing authentications for different purposes (how to auth to Providers, how to auth to VMs, how to auth to Hosts, etc...).  I think Container Deployments having its own Authentications table will complicate things more than simplify.

And, it will help with refactoring later if/when we want to figure out better ways to build common Authentication storage mechanisms.